resources/web/wwi/proto_viewer.js

```diff
@@ -0,0 +1,188 @@
+/* global showdown */
+/* global hljs */
+
+function populateProtoViewDiv(mdContent, imgPrefix, infoArray) {
+  const view = document.getElementsByClassName('proto-doc')[0];
+  while (view.firstChild)
+    view.removeChild(view.firstChild);
+
+  // markdown to html
+  window.mermaidGraphCounter = 0;
+  window.mermaidGraphs = {};
+  const converter = new showdown.Converter({tables: 'True',
+    extensions: [
+      'wbTabComponent', 'wbSpoiler', 'wbChart', 'wbVariables', 'wbAPI', 'wbFigure', 'wbAnchors',
+      'wbIllustratedSection', 'youtube'
+    ]});
+  const html = converter.makeHtml(mdContent);
+  view.innerHTML = html;
+
+  setupModalWindow('#main-container');
+  renderGraphs();
+  redirectImages(view, imgPrefix);
+  updateModalEvents(view);
+  redirectUrls(view);
+  highlightCode(view);
+  view.insertBefore(infoArray, view.firstChild);
+}
+
+function renderGraphs() {
+  for (let id in window.mermaidGraphs) {
+    window.mermaidAPI.render(id, window.mermaidGraphs[id], function(svgCode, bindFunctions) {
+      document.querySelector('#' + id + 'Div').innerHTML = svgCode;
+      // set min-width to be 2/3 of the max-width otherwise the text might become too small
+      const element = document.querySelector('#' + id);
+      const style = element.getAttribute('style');
+      element.setAttribute('style',
+        style + ' min-width:' + Math.floor(0.66 * parseInt(style.split('max-width:')[1].split('px'))) + 'px;');
+    });
+  }
+}
+
+function highlightCode(view) {
+  const supportedLanguages = ['c', 'cpp', 'java', 'python', 'matlab', 'sh', 'ini', 'tex', 'makefile', 'lua', 'xml',
+    'javascript'];
+
+  for (let i = 0; i < supportedLanguages.length; i++) {
+    const language = supportedLanguages[i];
+    hljs.configure({languages: [ language ]});
+    const codes = document.querySelectorAll('.' + language);
+    for (let j = 0; j < codes.length; j++) {
+      const code = codes[j];
+      hljs.highlightBlock(code);
+    }
+  }
+}
+
+function redirectUrls(node) {
+  // redirect a's href
+  const as = node.querySelectorAll('a');
+  for (let i = 0; i < as.length; i++) {
+    const a = as[i];
```

Comment:
probably can be avoided with a `for (const item of as)` loop instead, assuming it's a list